040-Typescript-SDK/130-ask.mdx

```diff
@@ -164,6 +159,29 @@ data: {"done":true,"records":["0415d654cc8dd113e9b9423f425697ae","91301cdc08524e
 
 You can find a full Next.js application using the ask endpoint with server-sent events in our [examples repository](https://github.com/xataio/examples/tree/main/apps/sample-chatgpt).
 
+## Asking follow-up questions
+
+Xata supports asking follow-up questions, enabling ChatGPT-style interfaces for your data. Here, we have the concept of a **session**. A session represents a single conversation between a user and data in one of your Xata tables. A session has data from your tables, aka **context**. It also persists the rules you defined when starting the session, and all question sent and answers received.
+
+When communicating with OpenAI, the ask endpoint ensures that your rules, your context, and as many questions as possible are transmitted each time. If the total amount of messages becomes too long for OpenAI to process, Xata will remove messages from early in the conversation until the message size is small enough to work.
+
+Each session maintains its own cached set of your context, so changes to the underlying data in your database will not affect ongoing conversations. All cached information: your context, rules, and all conversations are automatically removed after 7 days.
+
+<TabbedCode tabs={['TypeScript', 'JSON']}>
+
+```ts
+const result = await xata.db.Tutorial.askTableContinue('<session_id>', '<your follow-up message>');
```

Comment:
@SferaDev I wrote this up to get the document complete, but i'm 99% sure we can find a better name here. WDYT?